aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -297,6 +297,62 @@ std::tuple<Tensor, Tensor, Tensor> _svd_helper_cuda_lib(const Tensor& self, bool
   return std::make_tuple(U_working_copy, S_working_copy, VT_working_copy);
 }
 
+
+template<typename scalar_t>
+inline static void apply_cholesky_lib_potrf(Tensor& self_working_copy, bool upper, Tensor& infos) {
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  const auto uplo = upper ? CUBLAS_FILL_MODE_UPPER : CUBLAS_FILL_MODE_LOWER;
+  const int n = cuda_int_cast(self_working_copy.size(-1), "n");
+  const int lda = std::max<int>(1, n);
+
+  at::cuda::solver::potrf<scalar_t>(
+    handle, uplo, n, self_working_copy.data_ptr<scalar_t>(), lda, infos.data_ptr<int>()
+  );
+}
+
+template<typename scalar_t>
+inline static void apply_cholesky_lib_potrfBatched(Tensor& self_working_copy, bool upper, Tensor& infos) {
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  const auto uplo = upper ? CUBLAS_FILL_MODE_UPPER : CUBLAS_FILL_MODE_LOWER;
+  const int n = cuda_int_cast(self_working_copy.size(-1), "n");
+  const int lda = std::max<int>(1, n);
+
+  const int batch_size = cuda_int_cast(batchCount(self_working_copy), "batch_size");
+  const scalar_t* self_data = self_working_copy.data_ptr<scalar_t>();
+  const int matrix_stride = cuda_int_cast(matrixStride(self_working_copy), "matrix_stride");
+
+  // cusolver batched kernels require input be "device array of device pointers"
+  Tensor self_working_copy_array = at::arange(
+    reinterpret_cast<int64_t>(self_data),
+    reinterpret_cast<int64_t>(&self_data[(batch_size-1) * matrix_stride]) + 1,
```

Comment:
this seems needlessly complicated, `self_data+batch_size*matrix_size` should be enough? It will also correctly produce empty arange for an empty batch.

Reply:
Fixed.